### ISSUE TITLE
Cannot assign value of declared type to descriptor on class instance

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1188]
**Describe the bug**
An error is produced for descriptor assignments where the `__set__` method is annotated with a type variable.  This used to work before 1.1.87.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `foo.bar = ""` and `foo.baz = [1]` would not generate an error.

**Screenshots or Code**
```python
# pyright: strict

from __future__ import annotations

from typing import Generic, TypeVar


_T = TypeVar("_T")


class Column(Generic[_T]):
    def __set__(self, instance: object, value: _T) -> _T:
        ...


class Foo:
    bar: Column[str] = Column()
    baz: Column[list[int]] = Column()

foo = Foo()

# Cannot assign member "bar" for type "Foo"
#   Type "Literal['']" cannot be assigned to type "_T"
foo.bar = ""

# Cannot assign member "baz" for type "Foo"
#   Type "List[int]" cannot be assigned to type "_T"
foo.baz = [1]

```

**VS Code extension or command-line**
Either, 1.1.87

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.88, which I just published. It will also be included in the next release of Pylance.
